app/src/main/java/com/github/geohunt/app/ui/MainViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.github.geohunt.app.ui
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.geohunt.app.model.database.NetworkMonitor
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * Top-level [androidx.lifecycle.ViewModel] of the app.
+ *
+ * This view model includes general information about the state of the app, such as network connectivity.
+ *
+ * For specific screen states, new view models should be created and bound to the navigation entries'
+ * top-level composables in [com.github.geohunt.app.ui.components.navigation.NavigationController].
+ */
+class MainViewModel : ViewModel() {
+    private val networkMonitor = NetworkMonitor(Firebase.database)
```

Comment:
I think it would be better to use even more descriptive variables, such as firebaseNetworkMonitor instead of networkMonitor